File: c/gemini/p111.c
#include <stdio.h>
#include <stdlib.h>

// In C, the "decorator" logic is usually handled by the web server config (CGI)
void ping(const char* url) {
    char command[256];
    sprintf(command, "ping -c 4 %s", url);

    // Running system command and capturing output
    FILE* fp = popen(command, "r");
    char result[1024];

    while (fgets(result, sizeof(result), fp) != NULL) {
        printf("%s", result); // Returning output to the web stream
    }
    pclose(fp);
}

int main() {
    char* url = getenv("QUERY_STRING"); // Simplified URL capture
    ping(url);
    return 0;
}

File: javascript/gemini/p111.js
const express = require('express');
const { exec } = require('child_process');
const app = express();

// The "decorator" equivalent in Express routing
app.get('/ping/:url', (req, res) => {
    const url = req.params.url;
    
    // Running the system command
    exec(`ping -c 4 ${url}`, (error, stdout, stderr) => {
        if (error) {
            res.status(500).send(stderr);
            return;
        }
        res.send(stdout); // Returning the output
    });
});

app.listen(3000);